pkg/tfbridge/provider.go

```diff
@@ -1013,15 +1012,21 @@ func (p *Provider) Create(ctx context.Context, req *pulumirpc.CreateRequest) (*p
 	label := fmt.Sprintf("%s.Create(%s/%s)", p.label(), urn, res.TFName)
 	glog.V(9).Infof("%s executing", label)
 
+	props, err := plugin.UnmarshalProperties(req.GetProperties(),
+		plugin.MarshalOptions{Label: label, KeepUnknowns: true, SkipNulls: true})
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshaling %s's new property state", urn)
+	}
 	// To get Terraform to create a new resource, the ID must be blank and existing state must be empty (since the
 	// resource does not exist yet), and the diff object should have no old state and all of the new state.
-	config, assets, err := UnmarshalTerraformConfig(ctx,
-		p, req.GetProperties(), res.TF.Schema(), res.Schema.Fields,
-		fmt.Sprintf("%s.news", label))
+	inputs, assets, err := makeTerraformInputsWithOptions(
+		ctx, nil, nil, nil, props, res.TF.Schema(), res.Schema.Fields,
+		makeTerraformInputsOptions{Defaults: False(), TFDefaults: False(), MaxItemsOneDefaults: True()},
```

Comment:
This is a very large change here that seems unrelated to the goals of the PR.  UnmarshalTerraformConfig used to apply defaults and it doesn't do that anymore. Why is it needed? Why do we believe it is safe?

Reply:
I don't believe `UnmarshalTerraformConfig` applied defaults but I agree it's not obvious:

It called `MakeTerraformConfig`:
https://github.com/pulumi/pulumi-terraform-bridge/blob/54802d5b003cd46ff6163a3367f5b8815217eacf/pkg/tfbridge/schema.go#L1241

which does not set the Defaults in the conversion context:

https://github.com/pulumi/pulumi-terraform-bridge/blob/54802d5b003cd46ff6163a3367f5b8815217eacf/pkg/tfbridge/schema.go#L1219-L1223

so they should not get applied